orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/appengine/DeployAppEngineConfigurationTask.java

```diff
@@ -0,0 +1,129 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.providers.appengine;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+import com.netflix.spinnaker.orca.api.pipeline.RetryableTask;
+import com.netflix.spinnaker.orca.api.pipeline.TaskResult;
+import com.netflix.spinnaker.orca.api.pipeline.models.ExecutionStatus;
+import com.netflix.spinnaker.orca.api.pipeline.models.StageExecution;
+import com.netflix.spinnaker.orca.clouddriver.KatoService;
+import com.netflix.spinnaker.orca.clouddriver.model.TaskId;
+import com.netflix.spinnaker.orca.clouddriver.tasks.AbstractCloudProviderAwareTask;
+import com.netflix.spinnaker.orca.pipeline.util.ArtifactUtils;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import javax.annotation.Nonnull;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class DeployAppEngineConfigurationTask extends AbstractCloudProviderAwareTask
+    implements RetryableTask {
+
+  private final ObjectMapper objectMapper;
+  private final KatoService kato;
+  private final String CLOUD_OPERATION_TYPE = "deployAppEngineConfiguration";
+  private final String CLOUD_PROVIDER = "appengine";
+  private final ArtifactUtils artifactUtils;
+
+  @Override
+  public long getBackoffPeriod() {
+    return 30000;
+  }
+
+  @Override
+  public long getTimeout() {
+    return 300000;
+  }
+
+  @Nonnull
+  @Override
+  public TaskResult execute(@Nonnull StageExecution stage) {
+    Map<String, Object> context = stage.getContext();
+    String account = getCredentials(stage);
+
+    if (context.get("cronArtifact") == null
+        && context.get("dispatchArtifact") == null
+        && context.get("indexArtifact") == null
+        && context.get("queueArtifact") == null) {
+      throw new IllegalArgumentException("At least one configuration artifact must be supplied.");
+    }
+
+    if (context.get("cronArtifact") != null) {
+      ArtifactAccountPair artifactAccountPair =
+          objectMapper.convertValue(context.get("cronArtifact"), ArtifactAccountPair.class);
+      Artifact artifact =
+          artifactUtils.getBoundArtifactForStage(
+              stage,
+              artifactAccountPair.getId(),
+              objectMapper.convertValue(artifactAccountPair.getArtifact(), Artifact.class));
+      context.put("cronArtifact", artifact);
```

Comment:
I might create another variable called `operationDescription` or something. Reading this the first time, I thought you were going to push this `context` back into the stage context.

Reply:
good idea -- done!